ofp/match.go

```diff
@@ -294,7 +293,7 @@ func readAllXM(r io.Reader, xms *[]XM) (int64, error) {
 
 // ReadFrom implements io.ReaderFrom interface. It deserializes
 // the OpenFlow extensible match from the given reader.
-func (xm *XM) ReadFrom(r io.Reader) (n int64, err error) {
+func (xm *XM) ReadFrom(r io.Reader, hasPayload bool) (n int64, err error) {
```

Comment:
It would be nice to determine the necessity to read payload by the length field of OXM. _If it is not possible_ for some reason, I would recommend to create another method for passing this flag:
```go
func (xm *XM) readFrom(r io.Reader, hasPayload bool) (int64, error) { /* ... */ }

func (xm *XM) ReadFrom(r io.Reader) (int64, error) {
    return xm.readFrom(r, true)
}
```

This is necessary to make ```XM``` implement [io.ReaderFrom](https://golang.org/pkg/io/#ReaderFrom) interface.